test/integration_test.py

```diff
@@ -995,49 +994,52 @@ def _file_ext(self, file: JSON) -> str:
         suffixes = PurePath(file['name']).suffixes
         return ''.join(suffixes).lower()
 
-    def _validate_file_content(self, content: ReadableFileObject, file_ext: str):
+    def _validate_file_content(self, content: ReadableFileObject, file: JSON):
+        file_ext = self._file_ext(file)
         if file_ext == '.fastq':
             self._validate_fastq_content(content)
         elif file_ext == '.fastq.gz':
             with gzip.open(content) as buf:
                 self._validate_fastq_content(buf)
-        else:
+        elif file['size'] > 0:
             self.assertEqual(1, len(content.read(1)))
+        else:
+            log.info('File %r is empty', file['uuid'])
 
     def _validate_file_response(self,
                                 response: urllib3.HTTPResponse,
-                                file_ext: str):
+                                file: JSON):
```

Comment:
Let's use a `TypedDict` with only the keys we use in the IT (`total=False`).

Reply:
I am not certain that `total=False` is appropriate here. [The docs](https://docs.python.org/3/library/typing.html#typing.TypedDict) indicate that this parameter allows for defined keys to be absent, rather than for undefined keys to present (as I infer you were intending).

Reply:
Agreed at PL to leave off `total=False`